tests/dnfserver/container/nginx.conf

```diff
@@ -0,0 +1,70 @@
+server {
+    listen       8443 ssl;
+    listen       [::]:8443 ssl;
+    server_name  localhost;
+    root         /dnfrepo;
+
+    ssl_certificate "/etc/pki/nginx/server.crt";
+    ssl_certificate_key "/etc/pki/nginx/server.key";
+
+    # enable client cert verification
+    ssl_client_certificate /etc/pki/myCA.crt; ## Use your own trusted certificate from CA/SSLTrust
+    ssl_verify_client on;
+
+    ssl_session_cache shared:SSL:1m;
+    ssl_session_timeout  10m;
+    ssl_ciphers PROFILE=SYSTEM;
+    ssl_prefer_server_ciphers on;
+    ssl_verify_depth 2;
+
+    # if you need to debug
+    error_log /var/log/nginx/error.log debug;
+
+
+    error_page 404 /404.html;
+        location = /40x.html {
+    }
+
+    error_page 500 502 503 504 /50x.html;
+        location = /50x.html {
+    }
+}
+
+server {
+    listen       8082;
+    listen       [::]:8082;
+    server_name  localhost;
+    root         /certificates;
+
+    # if you need to debug
+    error_log /var/log/nginx/error.log debug;
+
+
+    error_page 404 /404.html;
+        location = /40x.html {
+    }
+
+    error_page 500 502 503 504 /50x.html;
+        location = /50x.html {
+    }
```

Comment:
I can drop it if we are ok committing the certs to the repo. Otherwise it is serving those certs so they can be fetched from the container.

Reply:
I don't mind checking certs in the repo, provided:
- it's all base64 encoded data
- the header is generic enough not to be affiliated to any entity
- have essentially unlimited validity

I would even be okay with us generating the set of certs on demand with each CI run and using less entropy to decrease the performance penalty, it'd we fine in local executions because this integration test would not run very often, but it might potentially have an impact on the GH CI runners - these VMs are provisioned on demand and we don't know anything about them, e.g. what entropy source they use (i.e. a virtualized PRNG, or paravirtualized source from the host) and it might slow the execution down, I don't know, so I tracking test certs in the repo is fine with me.

Reply:
That said, we can leave out CSR, SRL and CA private key, the certs won't be maintained in any way and if a change is needed, they'll be regenerated from scratch.